core/src/App.svelte

```diff
@@ -897,26 +893,51 @@
   setContext('getUnsavedChangesModalPromise', getUnsavedChangesModalPromise);
 
   //// MICRO-FRONTEND MODAL
+  // list containing the opened modals
+  let mfModalList = [];
 
-  let mfModal = {};
-
-  const resetMicrofrontendModalData = () => {
-    mfModal.displayed = false;
-    mfModal.nodepath = undefined;
-    mfModal.settings = {};
-    modalIframe = undefined;
+  /**
+   * Resets the mf modal data given the index and updates the 'mfModalList'. If no index given, resets the whole list instead
+   * @param index {number|undefined}  the index of the modal to reset
+   */
+  const resetMicrofrontendModalData = (index) => {
+    if (index !== undefined) {
+      let tempMfModalList = mfModalList;
+      // use splice to remove the modal element and re-index the list.
+      tempMfModalList.splice(index, 1);
+      // svelte components rendered from a list, expect list changes to be done through variable re-assignment
+      mfModalList = tempMfModalList;
+    } else {
+      // reset all modal list
+      mfModalList = [];
+    }
   };
 
   resetMicrofrontendModalData();
 
+  /**
+   * Opens the (iframe/wc) view in a modal given in the nodepath 
+   * @param nodepath {string} the path of the view to open
+   * @param settings {Object} the respective modal settings
+   */
   const openViewInModal = async (nodepath, settings) => {
+    // check if navigation to this path is allowed or not
     if (await NavigationHelpers.shouldPreventNavigationForPath(nodepath)) {
       return;
     }
-    mfModal.displayed = true;
-    mfModal.nodepath = nodepath;
-    mfModal.settings = settings;
 
+    // insert modal into the modals list to be viewed on top of other modals
+    const tempModalList = mfModalList;
+    tempModalList.push({
+      mfModal : {
+        displayed: true,
+        nodepath,
+        settings
+      }
+    });
+    mfModalList = tempModalList;
```

Comment:
```suggestion
    const newModal = {
      mfModal : {
        displayed: true,
        nodepath,
        settings
      }
    };
    mfModalList = [ ...mfModalList, newModal ];
```
I suggest go for es6 way instead of an extra temp variable. Please check if it works fine.

Reply:
Good idea. I had some problems with this part but I'll give this suggestion a try 👍